### PR TITLE
getting started guide changes part 1

### PR DESCRIPTION
**Pull request by codemonkey9000**
This is part 1 of a 2 part overhaul to the getting started guide. This PR includes content changes and comments added to 3/5 files.

This was done to make it easier to review since the bulk of the changes will be made in the remaining 2 files.

### CODE REVIEW

File: source/includes/getting-started/_concepts.html.md [hunk 1]
```diff
@@ -1,61 +1,69 @@
...
 ![The Bonsai Platform][1]
 
 ## Build
 
-* **CLI:** the Bonsai Command Line Interface (CLI) tool.
-* **brains.bons.ai:** The address for Bonsai's web interface; this is the browser based tool that includes your BRAIN Dashboard and BRAIN Details page.
-* **Simulators:** are a training source. They imitate a real world situation in a virtual environment. Simulators are often a complex, interactive environment. They usually have state, which is a representation of the world inside the virtual environment.
+* **Bonsai Command Line Interface (CLI):** is a command line tool that enables you to configure and control the Bonsai Artificial Intelligence (AI) Engine.
+* **beta.bons.ai:** The address for Bonsai's beta web interface; this is the browser based tool that includes your *BRAIN Dashboard* and *BRAIN Details* page.
+* **Simulators:** are currently the only training source available. Simulators are often a complex, interactive virtual environment. They usually have state, which is a representation of the world inside the virtual environment.
```
Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: "are currently the only training source **implemented**"

File: source/includes/getting-started/_concepts.html.md [hunk 2]
```diff
@@ -1,61 +1,69 @@
...
-* **Data:** is a training source. It is information related to the concepts being trained that are used for training or test and evaluate how well the training is implemented. Non-generated data may sometimes be difficult to find, and the data that exists may be a very small set or difficult to label. An example of a data set you could train with is the MNIST database, which is a collection of handwritten digits used for training image processing systems and other machine learning systems.
+* **Data:** is another training source. It is information related to the concepts being trained that are used for training or test and evaluate how well the training is implemented. Non-generated data may sometimes be difficult to find, and the data that exists may be a very small set or difficult to label. An example of a data set you could train with is the MNIST database, which is a collection of handwritten digits used for training image processing systems and other machine learning systems.
```
Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: - "is **also a** training source."
- you mention `concept` here, but that term hasn't been introduced yet.  is there a better way to frame this?
- "that are used for training **, or testing and evaluating** how well the training is implemented."
- add a link to the MNIST database mentioned

File: source/includes/getting-started/_concepts.html.md [hunk 3]
```diff
@@ -1,61 +1,69 @@
...
 
 **Note:** Currently, during our private beta, you can only train with simulators as your training source.
```
Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: "you can only **use** simulators as your training source."
(having different conjugations of train is jarring)

File: source/includes/getting-started/_concepts.html.md [hunk 4]
```diff
@@ -1,61 +1,69 @@
...
 
-To learn more about connecting your own simulator, refer to our SDK.
+Today, we only support simulators written in python. To learn more about connecting your own simulator, refer to our SDK Reference (Coming soon!).
+
+[//]: # (Simulators only in Python, No SDK Reference)
 
 ## What is training?
 
-Training means teaching it to get a specific outcome on a simulator. The training involves using a specific set of concepts, a curriculum, and lessons, which are described in the Inkling file.
+Training means teaching a simulator to get a specific outcome. The training involves using a specific set of concepts, a curriculum, and lessons, which are described in the Inkling file.
```
Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: The BRAIN/AI is learning here, not the simulator.
"means teaching **a BRAIN** to reach a specific outcome **in the simulator**.

File: source/includes/getting-started/_concepts.html.md [hunk 5]
```diff
@@ -67,23 +75,28 @@ This means that you, the developer, can leverage your domain-specific expertise
...
+The Bonsai Command Line Interface (CLI) is a command line tool that enables you to configure the Bonsai AI Engine. The CLI is especially useful for automation and connection to other tools. Currently, there are some actions that can only be performed using the CLI, such as loading your Inkling file and connecting your simulator.
+
+To learn more about the Bonsai CLI, refer to our [CLI Reference][3].
 
 ## beta.bons.ai
 
-The Bonsai Beta Website is a browser-based tool for configuring and analyzing BRAINs stored in the Bonsai Artificial Intelligence Engine. The website will be useful for sharing, collaborating, and learning. There is some information you can only get from beta.bons.ai, such as a visualization of a BRAIN's training progress.
+The Bonsai Beta Website is a browser-based tool for configuring and analyzing BRAINs stored in the Bonsai AI Engine. In the future, this website will be useful for sharing BRAINs, collaborating, and learning. There is some information you can only get from beta.bons.ai, such as a visualization of a BRAIN's training progress.
```
Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: link to the website?

**Comment by codemonkey9000** [comment]
I made some changes to the heading structure while I was working on the changes that @jbeckman-bonsai suggested. I am 100% happier with the consistency, transparency, and flow of the new headings.

@keenbrowne let me know what you think of the new headings.

File: source/includes/getting-started/_concepts.html.md [hunk 6]
```diff
@@ -1,61 +1,67 @@
-# Architecture
+# Background FAQ
 
-There are some things you might want to know before you get started. In this section, we're going to walk you through the Bonsai Platform components and some of the basic Bonsai concepts. Let's jump right in!
+[//]: # (If any commented statements become false, change prior text)
+
+There are some things you might want to know before you get started. In this section, we're going to walk you through the Bonsai Platform components and some of the basic Bonsai concepts that are frequently asked questions. Let's jump right in!
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: The phrase "Bonsai concepts" is very easily confused with "Bonsai Concepts" (as in the concept keyword). Sam previously resolved this ambiguity by using the word "fundamentals"--fundamentals of the Bonsai Platform OR fundamental ideas of the Bonsai Platform

File: source/includes/getting-started/_concepts.html.md [hunk 7]
```diff
@@ -1,61 +1,67 @@
...
 ![The Bonsai Platform][1]
 
-## Build
+###### Build
 
-* **CLI:** the Bonsai Command Line Interface (CLI) tool.
-* **brains.bons.ai:** The address for Bonsai's web interface; this is the browser based tool that includes your BRAIN Dashboard and BRAIN Details page.
-* **Simulators:** are a training source. They imitate a real world situation in a virtual environment. Simulators are often a complex, interactive environment. They usually have state, which is a representation of the world inside the virtual environment.
+* **Bonsai Command Line Interface (CLI):** is a command line tool that enables you to configure and control the Bonsai Artificial Intelligence (AI) Engine.
+* **beta.bons.ai:** The address for Bonsai's beta web interface; this is the browser based tool that includes your *BRAIN Dashboard* and *BRAIN Details* page.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: beta.bons.ai should probably be a link so the user can click through

File: source/includes/getting-started/_concepts.html.md [hunk 8]
```diff
@@ -1,61 +1,67 @@
...
+* **Simulators:** are currently the only training source implemented. Simulators are often a complex, interactive virtual environment. They usually have state, which is a representation of the world inside the virtual environment.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: The word "training source" should be defined. Most tools currently only support batch data so the word "training source" will be new to many users.  A training source is any source of data used to teach or train an AI. It includes batch data like a large spreadsheet or a set of labels and images and synthetic data like the output from a simulation.

File: source/includes/getting-started/_concepts.html.md [hunk 9]
```diff
@@ -1,61 +1,67 @@
...
 * **Generators:** are a training source. They produce labeled data programmatically. This data is effectively infinite. A generator could, for example, produce a random (but known) integer, set of line segments, etc.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: Generators are not yet supported. (perhaps just adding that statement is sufficient)

File: source/includes/getting-started/_concepts.html.md [hunk 10]
```diff
@@ -1,61 +1,67 @@
...
-* **Data:** is a training source. It is information related to the concepts being trained that are used for training or test and evaluate how well the training is implemented. Non-generated data may sometimes be difficult to find, and the data that exists may be a very small set or difficult to label. An example of a data set you could train with is the MNIST database, which is a collection of handwritten digits used for training image processing systems and other machine learning systems.
+* **Data:** is also a training source. It is information related to the scenario being trained, or to test and evaluate how well the training is implemented. Non-generated data may sometimes be difficult to find, and the data that exists may be a very small set or difficult to label. An example of a data set you could train with is the [MNIST database][5], which is a collection of handwritten digits used for training image processing systems and other machine learning systems.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: This description makes me chuckle because it feels very pedantic--most users will already be familiar with the idea of data.  Perhaps:

**Data:** is also a training source. It is information related to the scenario being trained and can also be used to test and evaluate how well the training is implemented.  Data could be a collection of images and labels or  the rows of a spreadsheet. An example of a data set you could train with is the [MNIST database][5], which is a collection of handwritten digits used for training image processing systems and other machine learning systems.

File: source/includes/getting-started/_concepts.html.md [hunk 11]
```diff
@@ -1,61 +1,67 @@
...
+
+**Note:** Currently, during our private beta, you can only use simulators as your training source.
 
-**Note:** Currently, during our private beta, you can only train with simulators as your training source.
+[//]: # (Simulators are the only training source, beta.bons.ai is current website instead of brains.bons.ai)
 
-## Teach
+###### Teach
 
 * **Inkling:** our unique, special purpose, pedagogical programming language. Pedagogical programming is a new programming paradigm. It expresses the solution to a problem in terms of how you teach the computer to find the solution as opposed to how to calculate the solution.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: Should Pedagogical Programming also get it's own definition or is it okay to attach the definition to inkling?

File: source/includes/getting-started/_concepts.html.md [hunk 12]
```diff
@@ -1,61 +1,67 @@
...
+* **Bonsai Artificial Intelligence (AI) Engine:** abstracts away and automates the low-level mechanics of artificial intelligence. Each Inkling program developed is fed into the Bonsai AI Engine in order to generate and train appropriate models (referred to as BRAINs).
 
-## Use
+###### Use
 
 * **Mastermind:** the Bonsai AI application for building AI. This tool will contain our Integrated Development Environment (IDE). **Note:** This tool will not be available for the private beta.
-* **SDK & Libraries:** enables you to connect your own simulators to the Bonsai Artificial Intelligence Engine. It also enables you to integrate your application with your BRAIN
-* **HTTP API:** enables AI creators and AI users to interact with the Bonsai Artificial Intelligence Engine.
+* **SDK & Libraries:** enables you to connect your own simulators to the Bonsai AI Engine. They also enable you to integrate your application with your BRAIN.
+* **HTTP API:** enables AI creators and AI users to interact with the Bonsai AI Engine.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: This is not important for this diff but I recommend we rename HTTP API to API

File: source/includes/getting-started/_concepts.html.md [hunk 13]
```diff
@@ -1,61 +1,67 @@
...
 
-###  After your BRAIN is created and trained, you can use it with your app. Ultimately, our platform enables you to build AI agents that work (or play) for you without a deep knowledge of machine learning.
+After your BRAIN is created and trained, you can use it with your app. Ultimately, our platform enables you to build AI agents that work (or play) for you without a deep knowledge of machine learning.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: Some more detail here might be helpful. Something like:

After your BRAIN is created and trained, you can use it with your app by connecting your app to the API generated by the AI Engine using the Bonsai library specific to your app's programming language. Ultimately, our platform enables you to build AI agents that work (or play) for you without a deep knowledge of machine learning.

File: source/includes/getting-started/_concepts.html.md [hunk 14]
```diff
@@ -67,23 +73,30 @@ This means that you, the developer, can leverage your domain-specific expertise
...
+The Bonsai Command Line Interface (CLI) is a command line tool that enables you to configure the Bonsai AI Engine. The CLI is especially useful for automation and connection to other tools. Currently, there are some actions that can only be performed using the CLI, such as loading your Inkling file and connecting your simulator.
 
-## What is the Bonsai Command Line Interface?
+To learn more about the Bonsai CLI, refer to our [CLI Reference][3].
 
-The Bonsai Command Line Interface (CLI) is a command line tool that enables you to configure the Bonsai Artificial Intelligence Engine. The CLI is especially useful for automation and connection to other tools. Currently, there are some actions that can only be performed using the CLI, such as loading your Inkling file and connecting your simulator.
+## What is beta.bons.ai for?
 
-## beta.bons.ai
+The [Bonsai Beta Website][4] is a browser-based tool for configuring and analyzing BRAINs stored in the Bonsai AI Engine. In the future, this website will be useful for sharing BRAINs, collaborating, and learning. There is some information you can only get from beta.bons.ai, such as a visualization of a BRAIN's training progress.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: Would we want to usually have [beta.bons.ai](https://beta.bons.ai) be a link? It seems weird to see a URL and have it not be a link to me but I could be convinced otherwise

File: source/includes/getting-started/_concepts.html.md (hunk 12 above)
Comment (codemonkey9000) [inline source/includes/getting-started/_concepts.html.md]: Alright I've made a task for it in Maniphest, it should probably be part of the larger overhaul.

File: source/includes/getting-started/_concepts.html.md (hunk 8 above)
Comment (codemonkey9000) [inline source/includes/getting-started/_concepts.html.md]: I have now added a description to before the image where you first see the words "training source" and it's again described a bit differently in the simulator section.

File: source/includes/getting-started/_concepts.html.md (hunk 9 above)
Comment (codemonkey9000) [inline source/includes/getting-started/_concepts.html.md]: Data isn't supported either yet, right? There is a note at the bottom of the section that again reiterates that simulators are the only thing you can currently use during private beta.

File: source/includes/getting-started/_concepts.html.md (hunk 11 above)
Comment (codemonkey9000) [inline source/includes/getting-started/_concepts.html.md]: I don't think we're trying to teach people what words mean here, if someone (or most people) don't know they can look it up so it doesn't seem like we're crazy people making up words.

**Comment by keenbrowne** [comment]
In this case "Pedagogical Programming" is a new proper noun invented by Mark. Inkling is the world's first and only language for Pedigogical Programming. At the moment the idea of pedagogical programming is tied up with Inkling but one day, if we succeed, there may be other pedagogical programming languages.
That being said I expect the reader should know the definition of pedagogical and pedagogy. If not, they can look it up.





On Fri, Feb 3, 2017 at 5:33 PM -0800, "Katherine McAuliffe" <notifications@github.com> wrote:












@codemonkey9000 commented on this pull request.





In source/includes/getting-started/_concepts.html.md:
>  
-## Teach
+###### Teach
 
 * **Inkling:** our unique, special purpose, pedagogical programming language. Pedagogical programming is a new programming paradigm. It expresses the solution to a problem in terms of how you teach the computer to find the solution as opposed to how to calculate the solution.



I don't think we're trying to teach people what words mean here, if someone (or most people) don't know they can look it up so it doesn't seem like we're crazy people making up words.

**Comment by codemonkey9000** [comment]
Hey @keenbrowne I've made your changes. Let me know if you have more, disagree with any of my edits, or if this is good to go.

**Comment by codemonkey9000** [comment]
@keenbrowne I know you mentioned it was hard to see your comments vs. changes like in Phabricator. Here you can just look at the changes I made since your last review (my last commit).

https://github.com/BonsaiAI/bonsaiai.github.io/pull/26/commits/73c49f1560e987ad025cd2382a52fc30fa5c37e8

File: source/includes/getting-started/_concepts.html.md [hunk 15]
```diff
@@ -2,29 +2,29 @@
...
 
 **Note:** Currently, during our private beta, you can only use simulators as your training source.
 
 [//]: # (Simulators are the only training source, beta.bons.ai is current website instead of brains.bons.ai)
 
 ###### Teach
 
-* **Inkling:** our unique, special purpose, pedagogical programming language. Pedagogical programming is a new programming paradigm. It expresses the solution to a problem in terms of how you teach the computer to find the solution as opposed to how to calculate the solution.
+* **Inkling:** our unique, special purpose, pedagogical programming language. Pedagogical programming is a new teaching programming paradigm. It expresses the solution to a problem in terms of how you teach the computer to find the solution as opposed to how to calculate the solution.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: Pedagogical programming is a "programming paradigm" not a "teaching programming paradigm". I would revert this change.

File: source/includes/getting-started/_concepts.html.md [hunk 16]
```diff
@@ -33,7 +33,7 @@ This image displays all of the components of Bonsai's Platform. The Platform ena
 * **SDK & Libraries:** enables you to connect your own simulators to the Bonsai AI Engine. They also enable you to integrate your application with your BRAIN.
 * **HTTP API:** enables AI creators and AI users to interact with the Bonsai AI Engine.
 
-After your BRAIN is created and trained, you can use it with your app. Ultimately, our platform enables you to build AI agents that work (or play) for you without a deep knowledge of machine learning.
+After your BRAIN is created and trained, you can use it with your app by connecting your app to the API generated by the AI Engine using the Bonsai library specific to your app's programming language. Ultimately, our platform enables you to build AI agents that work (or play) for you without a deep knowledge of machine learning.
```
Comment (keenbrowne) [inline source/includes/getting-started/_concepts.html.md]: Should we use 'app' instead of 'application'?

**Comment by keenbrowne** [comment]
This looks good. I made a couple minor comments. Once we make a call on those minor points, this should be good to go.

File: source/includes/getting-started/_concepts.html.md (hunk 16 above)
Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: in my mind, 'app' == mobile.  i like application in this context since it encompasses everything.

Comment (jbeckman-bonsai) [inline source/includes/getting-started/_concepts.html.md]: Also, @keenbrowne, am I correct that the only SDK available is for Python?  Should that be called out here?

**Comment by jbeckman-bonsai** [comment]
Unless @keenbrowne has other input, this LGTM 👍
